src/components/Asset/AssetActions/Trade/TradeInput.tsx

```diff
@@ -61,11 +62,15 @@ export default function TradeInput({
             form={form}
             value={`${field.value}`}
             onChange={(e: ChangeEvent<HTMLInputElement>) => {
-              handleValueChange(name, Number(e.target.value))
-              validateForm()
               handleChange(e)
+              handleValueChange(name, Number(e.target.value))
+              // timeout needed to avoid validating the wrong (pass) value
```

Comment:
cool trick, remember we have some `lodash.debounce` usage in our codebase so search for that and mot likely this would be better package to do what you're trying to do here. Cause then we do not have to worry to clear that timeout anywhere

Reply:
great. Will do. 👍🏽